sqlmesh/core/config/connection.py

```diff
@@ -527,6 +527,9 @@ def _connection_factory(self) -> t.Callable:
 
         return connect
 
+    def get_catalog(self) -> t.Optional[str]:
```

Comment:
let's just make this a property catalog?

Reply:
This is in the connection config section and I don't want to create a base abstract property of `catalog` since a connection might itself have a field called `catalog`. Although it is likely that is what we want, that isn't certain. So I like an explicit `get_` pattern here since it makes it more clear that this may not even been a field of a given connection or even relevant to a connection (which it would return None).